be/src/exec/spill/dir_manager.cpp

```diff
@@ -14,25 +14,66 @@
 
 #include "exec/spill/dir_manager.h"
 
+#include <regex>
+
 #include "common/config.h"
 #include "storage/options.h"
+#include "storage/storage_engine.h"
+#include "storage/utils.h"
 
 namespace starrocks::spill {
 
 Status DirManager::init() {
-    std::vector<starrocks::StorePath> storage_paths;
-    RETURN_IF_ERROR(parse_conf_store_paths(config::spill_local_storage_dir, &storage_paths));
-    if (storage_paths.empty()) {
+    std::vector<starrocks::StorePath> spill_local_storage_paths;
+    RETURN_IF_ERROR(parse_conf_store_paths(config::spill_local_storage_dir, &spill_local_storage_paths));
+    if (spill_local_storage_paths.empty()) {
         return Status::InvalidArgument("cannot find spill_local_storage_dir");
     }
 
-    for (const auto& path : storage_paths) {
+    auto storage_paths = starrocks::StorageEngine::instance()->get_store_paths();
+    std::set<std::string> storage_path_set(storage_paths.begin(), storage_paths.end());
+
+    for (auto iter = spill_local_storage_paths.begin(); iter != spill_local_storage_paths.end();) {
+        const auto& path = iter->path;
+        if (storage_path_set.find(path) != storage_path_set.end()) {
+            return Status::InvalidArgument(fmt::format(
+                    "spill_local_storage_dir {} already exists in storage_root_path, please use another path", path));
+        }
```

Comment:
> if restart after crash, it will also start failed?

it depends on user's configuration, if spill_local_storage_dir is valid, if will start successfully. Whether it has crashed before does not affect whether it can be started successfully